send-transaction-service/src/send_transaction_service.rs

```diff
@@ -702,14 +706,22 @@ impl SendTransactionService {
 
             let iter = wire_transactions.chunks(config.batch_size);
             for chunk in iter {
+                let mut addresses = config
```

Comment:
Is this intentionally only implemented for tx retries? If not, we need to make changes inside `SendTransactionService::send_transactions_in_batch()` as well.
And yes, it is ridiculous that the initial send and retries don't use the same api so that this change only needs to be made once.
Refactoring doesn't look too awful; the main difference seems to be `debug!` logging with leader slot. But fine with me to kick that can.

Reply:
oh damn... i must've grepped poorly. nice catch!

Reply:
i went ahead and replicoded for this change set.